### ISSUE TITLE
feat: add support for multiple helmfile "environments" to be targeted on the CLI 

### ISSUE THREAD

**Issue by bitsofinfo** [roboll/helmfile#633]
Currently I can only do this

```
helmfile -f myhelmfile.yaml -e myenv1 apply
```

Would be great to be able to do this, where when multiple are present, they are merged into a single `.Environment`
```
helmfile -f myhelmfile.yaml -e app1,cluster1 apply
```
OR 
```
helmfile -f myhelmfile.yaml -e app1 -e cluster1 apply
```

Why? 

Consider this:

myhelmfile.yaml
```
bases:
  - environments.yaml
---
releases:
  - name: myapp
    namespace: default
    chart: stable/nginx-ingress
    values:
      -  someArgument: {{.Environment.cluster.name}}
          anotherArgument {{.Environment.app.setting1}}
```

environments.yaml
```
environments:
  app1:
    values:
      - app1.yaml
  app2:
    values:
      - app2.yaml
  ...
  app1000:
    values:
      - app1000.yaml
  cluster1:
     values:
       - cluster1.yaml
  cluster2:
     values:
       - cluster2.yaml
```

I have a handful of clusters, but hundreds of apps. Each release of an app definitely has "helmfile-environment" vars specific to an "app" but some of them while consumed by an app's chart, are more logically grouped and specific to a cluster. Being able to do it this way would let me be a bit more DRY.

**CURRENT WORKAROUND**

NOTE: currently I can achieve this as similar to the below, it works but not ideal as I have to set an ENV var then use that as a dynamic key to fetch it from a "clusters" dict (i.e. via sprig `values`)

```
export CLUSTER_NAME=cluster1; helmfile -f myhelmfile.yaml -e app1 apply
```

myhelmfile.yaml
```
bases:
  - environments.yaml
---

{{- $clusterEntry := pick .Environment.Values.clusters (env CLUSTER_NAME) }}
{{- $cluster := values $clusterEntry | first }}

releases:
  - name: myapp
    namespace: default
    chart: stable/nginx-ingress
    values:
      -  someArgument: {{$cluster.name}}
          anotherArgument {{.Environment.app.setting1}}
```

environments.yaml
```
environments:
  app1:
    values:
      - app1.yaml
      - allclusters.yaml
  app2:
    values:
      - app2.yaml
      - allclusters.yaml
  ...
  app1000:
    values:
      - app1000.yaml
      - allclusters.yaml
 
```

allclusters.yaml
```
clusters:
  cluster1:
     name: xxxxxx
     ...
   cluster2:
     ....
```

potentially related to #361

**Comment by mumoshu** [comment]
@bitsofinfo Thanks!

My understanding is that this use-case can be addressed with the latest proposal made in #361, with:

```
values:
- allclusters.yaml

environments:
  app1:
    values:
      - app1.yaml
  app2:
    values:
      - app2.yaml
  ...
  app1000:
    values:
      - app1000.yaml
```

For `app1`, `.Values` is `allclusters.yaml + app1.yaml`, wheres the latter merges into the former.

Am I correct?

**Comment by bitsofinfo** [comment]
perfect, then that will provide basically the same thing! cool
